components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/handlers/graphQL/analytics/GraphQLOperationInfoAnalyzer.java

```diff
@@ -0,0 +1,89 @@
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.apimgt.gateway.handlers.graphQL.analytics;
+
+
+import graphql.language.AstPrinter;
+import graphql.language.Definition;
+import graphql.language.Document;
+import graphql.language.Field;
+import graphql.language.OperationDefinition;
+import graphql.language.Selection;
+import graphql.parser.Parser;
+import graphql.schema.GraphQLSchema;
+import graphql.schema.idl.TypeDefinitionRegistry;
+import org.apache.synapse.MessageContext;
+import org.wso2.carbon.apimgt.common.gateway.graphql.GraphQLOperationAnalyzer;
+import org.wso2.carbon.apimgt.impl.APIConstants;
+
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+
+
+/**
+ * analyze each query and mutation
+ * then engages metric handler to handle metric publishing
+ */
+public class GraphQLOperationInfoAnalyzer {
+
+    private GraphQLAnalyticsMetricsHandler graphQLMetricsHandler;
+
+    public void analyzePayload(MessageContext messageContext){
+
+        Parser parser = new Parser();
+        String payload = (String) messageContext.getProperty(APIConstants.GRAPHQL_PAYLOAD);
+        Document document = parser.parseDocument(payload);
+        GraphQLSchema graphQLSchema = (GraphQLSchema) messageContext.getProperty(APIConstants.GRAPHQL_SCHEMA);
+        TypeDefinitionRegistry typeDefinition = (TypeDefinitionRegistry) messageContext.getProperty(APIConstants.TYPE_DEFINITION);
+        Map<String, Object> variablesMap = (Map<String, Object>) messageContext.getProperty(APIConstants.VARIABLE_MAP);
+        String complexityInfoJson = (String) messageContext
+                .getProperty(APIConstants.GRAPHQL_ACCESS_CONTROL_POLICY);
+        String type = null;
+
+        for (Definition definition : document.getDefinitions()) {
+            OperationDefinition operation = (OperationDefinition) definition;
+            type = operation.getOperation().toString().toLowerCase(Locale.ROOT);
+            for (Selection selection : operation.getSelectionSet().getSelections()) {
+                String result = AstPrinter.printAst(selection);
+                Document subDocument = parser.parseDocument( type + " { \n " + result + "\n }");
+
+                Map<String, Object> operationInfo = GraphQLOperationAnalyzer
+                        .getOperationInfo(type, selection, subDocument,graphQLSchema, typeDefinition, complexityInfoJson, variablesMap);
+                messageContext.setProperty(APIConstants.OPERATION_INFO, operationInfo);
+
+                List<Map> fieldUsage = GraphQLOperationAnalyzer.getUsedFields(selection);
+                messageContext.setProperty(APIConstants.ACCESSED_FIELDS, fieldUsage);
+
+                graphQLMetricsHandler = new GraphQLAnalyticsMetricsHandler();
+                if (type.equals(APIConstants.QUERY)){
```

Comment:
Swap the condition

Reply:
Fixed via https://github.com/wso2/carbon-apimgt/pull/12441/commits/226c49a261240dc5305f96d69aadb6f86edfae84